aten/src/ATen/native/hammerblade/OffloadIter.cpp

```diff
@@ -19,54 +19,53 @@ void offload_iterator_op_impl(TensorIterator& iter, std::vector<eva_t> device_sc
     TORCH_INTERNAL_ASSERT(iter.device(arg).is_hammerblade());
   }
 
-  // It is very important to use serial_for_each here, since we assume a single
-  // HammerBlade device in the system
-  iter.serial_for_each([&](char** data, const int64_t* strides, int64_t n) {
-    // Device pointers to tensors on the device
-    std::vector<eva_t> device_args;
-    std::vector<eva_t> device_ptrs;
-
-    // buffer for changing strides layout
-    int64_t *local_strides = (int64_t*) malloc(iter.ndim() * sizeof(int64_t));
-    if(!local_strides) {
-      AT_ERROR("Failed to allocate space for tmp strides on host");
-    }
+  // Device pointers to tensors on the device
+  std::vector<eva_t> device_args;
+  std::vector<eva_t> device_ptrs;
 
-    // Allocate device tensors and copy the data
-    for(int i=0; i<iter.ntensors(); i++) {
-      // ------------------------------
-      // strides of iter are stored as:
-      // dim0: tensor0, tensor1, tensor2
-      // dim1: tensor0, tensor1, tensor2
-      // ------------------------------
-      // we want per tensor strides:
-      // tensor0: dim0, dim1
-      // ------------------------------
-      for(int j=0; j<iter.ndim(); j++) {
-        local_strides[j] = (uint32_t)strides[i + j * ntensors];
-      }
-      // Iterate over all tensors to create
-      // corresponding tensors on the device.
-      eva_t device_arg = create_device_tensor(n, iter.ndim(),
-          (const int64_t*)local_strides, NULL, data[i], device_ptrs);
-      device_args.push_back(device_arg);
-    }
+  auto N = (uint32_t)iter.numel();
+
+  auto shape = iter.shape();
+
+  int64_t* sizes = (int64_t*) malloc(iter.ndim() * sizeof(int64_t));
+  TORCH_INTERNAL_ASSERT(sizes, "HB memory allocation failed on host");
+  for(int i = 0; i < iter.ndim(); ++i) {
+    sizes[i] = (int64_t) shape[i];
+  }
 
-    // free strides buffer
-    free(local_strides);
+  // buffer for changing strides layout
+  int64_t *local_strides = (int64_t*) malloc(iter.ndim() * sizeof(int64_t));
+  TORCH_INTERNAL_ASSERT(local_strides, "HB memory allocation failed on host");
 
-    // Add device scalars to arugments
-    for(int i=0; i<device_scalars.size(); i++) {
-      auto alpha = device_scalars[i];
-      device_args.push_back(alpha);
+  // Allocate device tensors and copy the data
+  for(int i=0; i<iter.ntensors(); i++) {
+
+    auto iter_strides = iter.strides(i);
+    // XXX: strides are measured in bytes
+    for(int j=0; j<iter.ndim(); j++) {
+      local_strides[j] = (int64_t)iter_strides[j];
```

Comment:
Think, iterator strides are in bytes to handle other data types like `bfloat16`. Could we just offload word strides so that we can only deal with words on device?

Did the same in reduction offloading... do you think we'd miss anything doing this?
https://github.com/cornell-brg/hb-pytorch/blob/ef8faba6523deb1690a649300991175c7c60e626/aten/src/ATen/native/hammerblade/OffloadIter.h#L54-L57

Reply:
Yep. I saw your code in reduction offloading. 
I didn't change it in this PR, only because I still haven't figured out why does TensorIterator choose to use bytes as strides measurement. ;)

Reply:
I think it's functionally correct because we'd be dividing with `sizeof(scalar_t)`. But yeah, the purpose of byte strides is unclear...